File: ex6_odd_even/src/MutualRecursion.fs
open System

let rec isOdd (n: int): bool =
        if n = 0 then
            false
        else
            isEven(n - 1)
    and isEven (n: int): bool =
        if n = 0 then
            true
        else
            isOdd(n - 1)

[<EntryPoint>]
let main argv =
    printfn "%b" (isOdd 10)
    printfn "%b" (isEven 10)
    printfn "%b" (isOdd 9)
    printfn "%b" (isEven 9)
    printfn "%b" (isOdd 2147483647)
    0

File: ex6_odd_even/src/MutualRecursion.scala
object MutualRecursion {
    def main(args: Array[String]): Unit = {
        println(isOdd(10))
        println(isEven(10))
        println(isOdd(9))
        println(isEven(9))
        println(isOdd(2147483647))
    }

    def isOdd(n: Int): Boolean = {
        if (n == 0)
            return false
        return isEven(n - 1)
    }

    def isEven(n: Int): Boolean = {
        if (n == 0)
            return true
        return isOdd(n - 1)
    }
}
